src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php

```diff
@@ -54,13 +55,32 @@ public function getConfigTreeBuilder(): TreeBuilder
         $rootNode = $tb->getRootNode();
 
         $rootNode
+            ->beforeNormalization()
+                ->always()
+                ->then(function ($v) {
+                    if (isset($v['hide_user_not_found']) && !isset($v['expose_security_errors'])) {
+                        $v['expose_security_errors'] = $v['hide_user_not_found'] ? ExposeSecurityLevel::None : ExposeSecurityLevel::All;
+                    } elseif (!isset($v['expose_security_errors'])) {
+                        $v['expose_security_errors'] = ExposeSecurityLevel::None;
+                    }
+
+                    return $v;
+                })
+            ->end()
             ->children()
                 ->scalarNode('access_denied_url')->defaultNull()->example('/foo/error403')->end()
                 ->enumNode('session_fixation_strategy')
                     ->values([SessionAuthenticationStrategy::NONE, SessionAuthenticationStrategy::MIGRATE, SessionAuthenticationStrategy::INVALIDATE])
                     ->defaultValue(SessionAuthenticationStrategy::MIGRATE)
                 ->end()
-                ->booleanNode('hide_user_not_found')->defaultTrue()->end()
+                ->booleanNode('hide_user_not_found')
+                    ->defaultNull()
+                    ->setDeprecated('symfony/security-bundle', '7.3', 'The "%node%" option is deprecated and will be removed in 8.0. Use the "expose_security_errors" option instead.')
+                ->end()
+                ->enumNode('expose_security_errors')
```

Comment:
You need to add the `expose-security-errors` attribute in the XSD as well, to make this setting usable in XML config files.